app/models/profile.rb

```diff
@@ -34,7 +21,33 @@ def activate
     end
   end
 
-  def verified?
-    verified_at.present?
+  def plain_pii
+    @_plain_pii ||= Pii::Attributes.new
+  end
+
+  def decrypt_pii(password)
+    Pii::Attributes.new_from_encrypted(encrypted_pii, password)
+  end
+
+  def encrypt_pii(password, pii = plain_pii)
+    ssn = pii.ssn
+    encryptor = Pii::Encryptor.new
+    self.ssn_signature = Digest::SHA256.hexdigest(encryptor.sign(ssn)) if ssn
```

Comment:
in my local tests, given a copy of the private key, it can take 1000+ hours of CPU time to create a rainbow table of all possible SSN signatures. That's pretty trivial from an attacker POV, but non-trivial when it comes time to rotate our private key.

When we rotate keys, since our `ssn_signature` is a one-way hash, we can't know what the original SSN is w/o building a lookup (rainbow) table and matching the existing signature. If we do that look up, we can then re-sign+save with the new key.

So key rotation seems like one major flaw with this design. I'd love to know if there are others, or if I've misidentified this one.